### PR TITLE
[docs] - supercollider/supercollider - supercollider/HelpSource/Classes/Server.schelp - minor attempts at fixing some typos and grammar

### PR DESCRIPTION
**Pull request by tmm2018**
#### [docs] - supercollider/supercollider - supercollider/HelpSource/Classes/Server.schelp - minor attempts at fixing some typos and grammar

below follows an attempt at:
- improving the quality of Server.schelp documentation file

my atttempt goes in the the direction of fixing issues related to:
- typing
- words mispelling
- grammar
- syntax
- puntuation
- other issues

---

i acknowledge that:
- the document may need some bugs to be fixed;
- my contribution may need some revisions;
- i may have accidentaly broken some community guidelines (currently contrbuting to several projects);

if that is the case:
- be honest
- don't hesitate in telling me what you think
- i will make sure i will fix everything to meet our standards

looking forward
kind regards/ best wishes
T.

---

</img></img>

### CODE REVIEW

File: HelpSource/Classes/Server.schelp [hunk 1]
```diff
@@ -5,7 +5,7 @@ related:: Classes/ServerOptions, Reference/Server-Architecture, Reference/Server
 
 description::
 
-A Server object is a representation of a server application. It is used to control scsynth (or supernova) from the SuperCollider language. (See link::Guides/Server-Guide::, as well as link::Guides/ClientVsServer:: for more details on the distinction.) It forwards OSC messages and has a number of allocators that keep track of IDs for nodes, buses and buffers.
+A Server object is a representation of a server application. It is used to control scsynth (or supernova) from the SuperCollider language. (See link::Guides/Server-Guide::, as well as a link::Guides/ClientVsServer:: for more details on the distinction.) It forwards OSC messages and has a number of allocators that keep track of IDs for nodes, buses, and buffers.
```
Comment (telephon) [inline HelpSource/Classes/Server.schelp]: why "See Server-Guide, as well as "a" ClientVsServer"? I think you mistake the `link::` tag for a word in the helpfile.

File: HelpSource/Classes/Server.schelp [hunk 2]
```diff
@@ -39,29 +39,29 @@ argument:: name
 a symbol;  each Server object is stored in one global classvariable under its name.
 
 argument:: addr
-an optional instance of link::Classes/NetAddr::, providing host and port.
+an optional instance of the link::Classes/NetAddr::, providing host and port.
```
Comment (telephon) [inline HelpSource/Classes/Server.schelp]: we usually say "an instance of `NetAddr`" and not: "an instance of the  `NetAddr`".

File: HelpSource/Classes/Server.schelp [hunk 3]
```diff
@@ -162,7 +162,7 @@ s.sendBundle(0.2, ["/s_new", "default", x = s.nextNodeID, 0, 1], ["/n_set", x, "
 method:: sendRaw
 
 method:: listSendMsg
-as sendMsg, but takes an array as argument.
+as sendMsg, but takes an array as an argument.
```
Comment (telephon) [inline HelpSource/Classes/Server.schelp]: we usually say "a method takes something as argument".

File: HelpSource/Classes/Server.schelp [hunk 4]
```diff
@@ -237,7 +237,7 @@ argument::onFailure
 A function that is called when quit has failed.
 
 argument::watchShutDown
-a boolean to tell the server whether to watch status during shutdown.
+a boolean to tell the server whether to watch status during the shutdown.
```
Comment (telephon) [inline HelpSource/Classes/Server.schelp]: "during shutdown" is correct.

File: HelpSource/Classes/Server.schelp [hunk 5]
```diff
@@ -273,7 +273,7 @@ Return an array mapping client IDs to their associated default groups as
 link::Classes/Group:: objects.
 
 method:: volume
-Get an instance of Volume that runs after the default group, or sets the Volume of the Server's output to level. Level is in db.
+Get an instance of Volume that runs after the default group, or sets the Volume of the Server's output to level. The level is in db.
```
Comment (telephon) [inline HelpSource/Classes/Server.schelp]: When we write "level is in db" we mean that the parameter is in db. This was correct.

**Comment by tmm2018** [comment]
hey @telephon 

thank you so much for your precious insight

i will make sure to:
- review the proposed changes today

looking forward
thank you so much
T.

**Comment by tmm2018** [comment]
dear @telephon fixed all your provided insight shall anyone provide any further feedback? please let me know what you think, and may we proceed with merging PR as soon further people have verified it

File: HelpSource/Classes/Server.schelp [hunk 6]
```diff
@@ -39,29 +39,29 @@ argument:: name
...
 
 method:: remote
-Create a new Server instance corresponding to a server app running on a separate machine. This method assumes the remote app has been booted, and starts listening immediately. You should not call link::#-boot:: on an instance created using this method.
+Create a new Server instance corresponding to a server app running on a separate machine. This method assumes the remote app has been booted and starts listening immediately. You should not call link::#-boot:: on an instance created using this method.
 
 argument:: name
 a symbol;  each Server object is stored in one global classvariable under its name.
 
 argument:: addr
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: Remove the before NetAddr. Add the before IP address and port.

File: HelpSource/Classes/Server.schelp [hunk 7]
```diff
@@ -91,14 +91,14 @@ Server.all
 ::
 
 method:: allRunningServers
-returns:: a Set containing all running servers, according to the definition of link::#-serverRunning::.
+returns:: a Set containing all running servers, according to the definition of the link::#-serverRunning::.
 
 code::
 Server.allRunningServers
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: Remove the before serverrunning

File: HelpSource/Classes/Server.schelp [hunk 8]
```diff
@@ -91,14 +91,14 @@ Server.all
...
 ::
 
 method:: allBootedServers
-returns:: a Set containing all booted servers, according to the definition of link::#-hasBooted::.
+returns:: a Set containing all booted servers, according to the definition of the link::#-hasBooted::.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: Remove the before hasBooted

File: HelpSource/Classes/Server.schelp [hunk 9]
```diff
@@ -153,7 +153,7 @@ method:: sendBundle
 send an OSC-bundle to the server.
 discussion::
 Since the network may have irregular performance, time allows for the bundle to be evaluated at a specified point in the future.
-Thus all messages are synchronous relative to each other, but delayed by a constant offset.
+Thus all messages are synchronous relative to each other but delayed by a constant offset.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: This is more a question of style than correctness, but personally I prefer it with the comma

File: HelpSource/Classes/Server.schelp [hunk 10]
```diff
@@ -222,10 +222,10 @@ If true, start a Routine to send a /status message to the server every so often.
 argument:: recover
 If true, create a new node ID allocator for the server, but use the old buffer and bus allocators. This is useful if the server process did not actually stop. In normal use, the default value "false" should be used.
 argument:: onFailure
-In this method, the onFailure argument is for internal use only. If you wish to take specific actions when the server boots or fails to boot, it is recommended to use link::#-waitForBoot:: or link::#-doWhenBooted::.
+In this method, the onFailure argument is for internal use only. If you wish to take specific actions when the server boots or fails to boot, it is recommended to use the link::#-waitForBoot:: or link::#-doWhenBooted::.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: Remove the added the

File: HelpSource/Classes/Server.schelp [hunk 11]
```diff
@@ -222,10 +222,10 @@ If true, start a Routine to send a /status message to the server every so often.
...
 
 discussion::
-You cannot boot a server app on a remote machine, but you can initialise the allocators by calling this message.
+You cannot boot a server app on a remote machine, but you can initialize the allocators by calling this message.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: Do we have policy re British/Canadian vs US spellings?

File: HelpSource/Classes/Server.schelp [hunk 12]
```diff
@@ -255,7 +255,7 @@ method:: status
 query the server status
 
 method:: notify
-The server sends notifications, for example if a node was created, a 'tr' message from a link::Classes/SendTrig::, or a strong::/done:: action. if code::flag:: is set to false, these messages will not be sent to this client. The default is true. If true the server will respond with a clientID (scsynth only) which can be useful in multi-client situations. If this is different from any previously received ID new allocators will be created. See link::#Local vs. Remote Servers, Multi-client Configurations:: for more information.
+The server sends notifications, for example, if a node was created, a 'tr' message from a link::Classes/SendTrig::, or a strong::/done:: action. if code::flag:: is set to false, these messages will not be sent to this client. The default is true. If true the server will respond with a clientID (scsynth only) which can be useful in multi-client situations. If this is different from any previously received ID new allocators will be created. See link::#Local vs. Remote Servers, Multi-client Configurations:: for more information.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: If should be upper case after action

File: HelpSource/Classes/Server.schelp [hunk 13]
```diff
@@ -339,22 +339,22 @@ method:: numSynths
 Get number of running link::Classes/Synth::s.
 
 method:: numGroups
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: Remove the before groups

File: HelpSource/Classes/Server.schelp [hunk 14]
```diff
@@ -339,22 +339,22 @@ method:: numSynths
...
 method:: pid
-Get process ID of running server (if not internal).
+Get process ID of the running server (if not internal).
 
 method:: addr
 The network address of the server as a link::Classes/NetAddr::.
 
 method:: maxNumClients
-If known, the maximum number of clients allowed on the server. Otherwise, the value of link::Classes/ServerOptions#-maxLogins::, which is what will be requested after the server boots. This number is not guaranteed to be correct until link::#-serverRunning:: is code::true::.
+If known, the maximum number of clients allowed on the server. Otherwise, the value of the link::Classes/ServerOptions#-maxLogins::, which is what will be requested after the server boots. This number is not guaranteed to be correct until link::#-serverRunning:: is the code::true::.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: Remove the before true

File: HelpSource/Classes/Server.schelp [hunk 15]
```diff
@@ -489,7 +489,7 @@ set the number of shared control buses. Must be done before the internal server
 
 subsection:: Persistent Node Trees
 
-The class link::Classes/ServerTree:: can be used to store functions which will be evaluated after the server is booted, after all nodes are freed, and after cmd-. is pressed.
+The class link::Classes/ServerTree:: can be used to store functions which will be evaluated after the server is booted, after all, nodes are freed, and after cmd-. is pressed.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: Remove comma after all

File: HelpSource/Classes/Server.schelp [hunk 16]
```diff
@@ -621,12 +621,12 @@ x.free; // stop the synths
 // look in your recordings folder and you'll find a file named for this date and time
 ::
 
-Recording is done via an of link::Classes/Recorder:: - a server holds one instance implicitly.
+The recording is done via an of the link::Classes/Recorder:: - a server holds one instance implicitly.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: An instance of recorder

File: HelpSource/Classes/Server.schelp [hunk 17]
```diff
@@ -653,13 +653,13 @@ method:: pauseRecording
 Pauses recording. Can be resumed by executing record again.
 
 method:: stopRecording
-Stops recording, closes the file, and frees the associated resources.
+Stops recording closes the file and frees the associated resources.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: This was correct

File: HelpSource/Classes/Server.schelp [hunk 18]
```diff
@@ -653,13 +653,13 @@ method:: pauseRecording
...
 discussion::
 You must call this when finished recording or the output file will be unusable. Cmd-. while recording has the same effect.
 
 
 method:: recChannels
-Get/set the number of channels (int) to record. Is automatically set to the value of link::Classes/ServerOptions#-numOutputBusChannels:: when booting the server. Must be called strong::before:: prepareForRecord.
+Get/set the number of channels (int) to record. Is automatically set to the value of the link::Classes/ServerOptions#-numOutputBusChannels:: when booting the server. Must be called strong::before:: prepareForRecord.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: This was correct

File: HelpSource/Classes/Server.schelp [hunk 19]
```diff
@@ -672,24 +672,24 @@ Get/set the size of the link::Classes/Buffer:: to use with the link::Classes/Dis
...
 subsection:: Asynchronous Commands
 
-Server provides support for waiting on the completion of asynchronous OSC-commands such as reading or writing soundfiles. N.B. The following methods must be called from within a running link::Classes/Routine::. Explicitly passing in a link::Classes/Condition:: allows multiple elements to depend on different conditions. The examples below should make clear how all this works.
+The server provides support for waiting on the completion of asynchronous OSC-commands such as reading or writing sound files. N.B. The following methods must be called from within a running link::Classes/Routine::. Explicitly passing in a link::Classes/Condition:: allows multiple elements to depend on different conditions. The examples below should make clear how all this works.
 
 method:: bootSync
 Boot the Server and wait until it has completed before resuming the thread.
 argument:: condition
-an optional instance of link::Classes/Condition:: used for evaluating this.
+an optional instance of the link::Classes/Condition:: used for evaluating this.
```
Comment (muellmusik) [inline HelpSource/Classes/Server.schelp]: Remove added the

Comment (muellmusik) [reply HelpSource/Classes/Server.schelp]: Also in similar cases below

File: HelpSource/Classes/Server.schelp (hunk 6 above)
Comment (tmm2018) [reply HelpSource/Classes/Server.schelp]: fixed

**Comment by tmm2018** [comment]
spent some extra time fixing some other issues that needed to be fixed, among them:
- unnecessary pronouns before class linking, et. al;
- unnecessary space modification

I wasn't able to fix the space vs tab in line 471, I guess, so if someone can fix that for me, I would be quite thankful shall we proceed with merging, or shall we wait for others to give further feedback. @telephon, please provide some feedback, if possible

looking forward
kind regards/best wishes
T.

**Comment by tmm2018** [comment]
by the way, is it possible to add the label docs to the PR? can someone please do that?

thank you so much

**Comment by jamshark70** [comment]
Tmm, can you please learn to use git so that you can address feedback with one or two commits instead of 20?

The commit history for this pull request is far out of proportion to the content changed. (True, it can be squash-merged later, but the current way also means you've generated literally 20 emails to every developer for a few lines worth of changes. That's really a lot. I don't want my inbox flooded for trivial documentation changes.)

IMO the Edit button in github is not the ideal way to do things. You should be editing local files on your machine, committing locally, and pushing to a branch that you own.

**Comment by tmm2018** [comment]
sure, I will start to commit to my local branch and then getting it to the central repository. I will try not to repeat the same mistake again. for some reason, I tend to work in this sort of way, so yeah, it might be not ok for you guys. I already have my own shell scripts for dealing with my very own repository managemente, but if i update them and try to get them a bit further, this will probably speed up the pace with which i make my own contributions, and making it more organized and also not so messy. thanks for the precious insight, @jamshark70 I will handle this when I arrive home

**Comment by tmm2018** [comment]
this is what I need towards updating my branch with the remote repo of the main supercollider distro, right? at least bearing in mind that this is a shell shortcut.

```
git remote -v && git remote add upstream $1 && git remote -v && git fetch upstream && git merge upstream/master && git push origin master
```

for pushing to my local repository, I shall use something like:
```
VAR1=$(pwgen 10 1) && VAR2=$(date +%Y-%m-%d) && VAR3="$1" && VARF=("$VAR2-$VAR1-$VAR3") && git pull  && git status && git add . && git status && git commit -m "$VARF"  && git status && git checkout -b "$VARF" && git push origin "$VARF" && git status && clear && git-issues
```

the git issues is a node.js formula that I installed from npm repository

then from that point, I make a pull request to the central repository

and then from that point bearing in mind that the repository I am pulling to a remote repository updating the branch where I created my repository should update the Pull Request. is that right?

**Comment by snappizz** [comment]
i have squash merged this so the 20 commits are combined into 1.

**Pull request merged**